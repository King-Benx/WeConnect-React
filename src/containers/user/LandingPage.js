import React from 'react';
import business_image from '../../images/business-3152586_640.jpg';
import { Link } from 'react-router-dom';
import { Button } from 'react-bootstrap';
import superagent from 'superagent';
import { BASE_URL } from '../../custom/constants';
import CustomFunctions from '../../custom/CustomFunctions';

class LandingPage extends React.Component{
    constructor(){
        super();
        this.state = { username:"", email:"",password:"",confirm_password:"" }
    }


    handleChange = (event) => {
        this.setState({ [event.target.id]: event.target.value });
    }
    
    handleReset = (event) => {
        this.setState({username:"", email:"",password:"",confirm_password:""});
    }

    formSubmit = (event) => {
        event.preventDefault();
        if (this.state.password === this.state.confirm_password){
            if (this.state.password.length > 4){
                superagent
                    .post(BASE_URL+'api/v1/auth/register')
                    .send({ username:this.state.username, email:this.state.email, password:this.state.password })
                    .end((err,res)=>{
                        if(err){ CustomFunctions.createNotifications(err.status, err.toString()) }; 
                        CustomFunctions.createNotifications(res.status,res.body.message);
                        this.props.history.replace('/login');
                    });
                }else{
                    CustomFunctions.createNotifications(400,'Password too short!');
                }
            }else{
                CustomFunctions.createNotifications(400,'Passwords are not similar');
        }
    }

    render(){
        return ( 
            <div className="row">
                <div className="col-lg-4 col-md-4 col-sm-6 col-xs-12 pull-right">
                    <div className="panel panel-info">
                        <div className="panel-heading">
                            <h3 className="panel-title">Registration Form</h3>
                        </div>
                        <div className="panel-body">
                            <form method="POST" onSubmit={ this.formSubmit } className="form">
                                <div className="form-group">
                                        <input type="text" className="form-control" placeholder="Username" value={this.state.username} id="username" name="username" onChange={ this.handleChange } required="required"/>
                                </div>
                                <div className="form-group">
                                    <input type="email" className="form-control" placeholder="johndoe@mail.com" value={this.state.email} id="email" name="email" onChange={ this.handleChange } aria-describedby="email-addon" required="required"/>
                                </div>
                                <div className="form-group">
                                    <input type="password" className="form-control" placeholder="Password" id="password" value={this.state.password}  name="password" onChange={ this.handleChange } aria-describedby="password-addon" required="required"/>
                                </div>
                                <div className="form-group">
                                    <input type="password" className="form-control" placeholder="Confirm Password" value={this.state.confirm_password} id="confirm_password"  onChange={ this.handleChange }  name="confirm_password" aria-describedby="confirm_password-addon"
                                            required="required"/>
                                </div>
                                <div className="form-group">
                                    <Button type="reset" onClick={ this.handleReset } className="pull-left"><i className="glyphicon glyphicon-refresh"></i> Reset</Button>
                                    <Button type="submit" className="pull-right" bsStyle="primary"><i className="glyphicon glyphicon-plus-sign"></i> Register</Button>
                                    <div className="clearfix"></div>
                                </div>
                            </form>
                            <div className="text-center">
                                <Link to="/login">Already a member? Login Here!</Link>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-lg-8 col-md-8 col-sm-6 col-xs-12 pull-left">
                    <div className="jumbotron">
                        <div className="container">
                            <h1>Welcome to WeConnect</h1>
                            <p>WeConnect provides a platform that brings businesses and individuals together. This platform creates
                                awareness for businesses and gives the users the ability to write reviews about the businesses
                                they have interacted with.</p>
                            <img src={ business_image } className="img img-responsive" alt=""/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default LandingPage;