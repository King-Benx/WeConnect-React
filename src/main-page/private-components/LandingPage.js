import React from 'react';
import business_image from '../img/business-3152586_640.jpg';
import {Link} from 'react-router-dom';
import { Button } from 'react-bootstrap';
import superagent from 'superagent';
import {BASE_URL} from '../functional-resources/urls';
import CustomFunctions from '../functional-resources/CustomFunctions';

class LandingPage extends React.Component{
    constructor(){
        super();
        this.state = {username:"", email:"",password:"",confirm_password:""}
    }


    handleChange = (event)=>{
        this.setState({[event.target.id]: event.target.value});
    }
    
    handleReset= (event)=>{
        this.setState({username:"", email:"",password:"",confirm_password:""});
    }

    formSubmit(event){
        event.preventDefault();
        if (this.state.password === this.state.confirm_password){
            if (this.state.password.length > 4){
                superagent
                    .post(BASE_URL+'api/v1/auth/register')
                    .send({username:this.state.username,email:this.state.email, password:this.state.password})
                    .end((err,res)=>{
                        if(err){CustomFunctions.createNotifications(err.status,err.toString())}; 
                        CustomFunctions.createNotifications(res.status,res.body.message);
                        this.props.history.replace('/login');
                    });
                }else{
                    CustomFunctions.createNotifications(400,'Password too short!');
                }
            }else{
                CustomFunctions.createNotifications(400,'Passwords are not similar');
        }
    }

    render(){
        return ( 
            <div className="row">
                <div className="col-lg-4 col-md-4 col-sm-6 col-xs-12 pull-right">
                    <div className="panel panel-info">
                        <div className="panel-heading">
                            <h3 className="panel-title">Registration Form</h3>
                        </div>
                        <div className="panel-body">
                            <form method="POST" onSubmit={this.formSubmit.bind(this)} className="form">
                                <div className="form-group">
                                    <div className="input-group">
                                        <span className="input-group-addon" id="username-addon">Username</span>
                                        <input type="text" className="form-control" placeholder="Username" value={this.state.username} id="username" name="username" onChange={this.handleChange.bind(this)} aria-describedby="username-addon" required="required"/>
                                    </div>
                                </div>
                                <div className="form-group">
                                    <div className="input-group">
                                        <span className="input-group-addon" id="email-addon">Email</span>
                                        <input type="email" className="form-control" placeholder="johndoe@mail.com" value={this.state.email} id="email" name="email" onChange={this.handleChange.bind(this)} aria-describedby="email-addon" required="required"/>
                                    </div>
                                </div>
                                <div className="form-group">
                                    <div className="input-group">
                                        <span className="input-group-addon" id="password-addon">Password</span>
                                        <input type="password" className="form-control" placeholder="Password" id="password" value={this.state.password}  name="password" onChange={this.handleChange.bind(this)} aria-describedby="password-addon" required="required"/>
                                    </div>
                                </div>
                                <div className="form-group">
                                    <div className="input-group">
                                        <span className="input-group-addon" id="confirm_password-addon">Confirm Password</span>
                                        <input type="password" className="form-control" placeholder="Confirm Password" value={this.state.confirm_password} id="confirm_password"  onChange={this.handleChange.bind(this)}  name="confirm_password" aria-describedby="confirm_password-addon"
                                            required="required"/>
                                        <span id='message'></span>
                                    </div>
                                </div>
                                <div className="form-group">
                                    <Button type="reset" onClick={this.handleReset.bind(this)} className="pull-left"><i className="glyphicon glyphicon-refresh"></i> Reset</Button>
                                    <Button type="submit" className="pull-right" bsStyle="primary"><i className="glyphicon glyphicon-plus-sign"></i> Register</Button>
                                    <div className="clearfix"></div>
                                </div>
                            </form>
                            <div className="text-center">
                                <Link to="/login">Already a member? Login Here!</Link>
                            </div>
                        </div>
                    </div>
                </div>
                <div className="col-lg-8 col-md-8 col-sm-6 col-xs-12 pull-left">
                    <div className="jumbotron">
                        <div className="container">
                            <h1>Welcome to WeConnect</h1>
                            <p>WeConnect provides a platform that brings businesses and individuals together. This platform creates
                                awareness for businesses and gives the users the ability to write reviews about the businesses
                                they have interacted with.</p>
                            <img src={business_image} className="img img-responsive" alt=""/>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
export default LandingPage;