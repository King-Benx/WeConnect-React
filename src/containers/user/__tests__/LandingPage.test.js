import React from 'react';
import { shallow } from 'enzyme';
import LandingPage from "../LandingPage";
import renderer from 'react-test-renderer';

describe('These are tests for the registration form for a user', () =>{
    
    const event={
        target:{
            value:{}
        },
        preventDefault: () => {

        }
    }

    let mountedLandingPage;

    beforeEach(() => {
        mountedLandingPage= shallow(<LandingPage/>);
    });


    it('handles input changed', () => {
        mountedLandingPage.instance().handleChange(event)
    })

    it('handles input reset', () => {
        mountedLandingPage.instance().handleReset(event)
    })

    // it('handles form submit', () => {
    //     mountedLandingPage.instance().formSubmit(event)
    // })

    it('renders without crashing', () => {
        shallow(<LandingPage />)
        const tree = renderer.create(<LandingPage />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('has a paragraph', () => {
        const paragraphs = mountedLandingPage.find('p')
        expect(paragraphs.length).toBe(1)
    });

    it('has a i', () => {
        const i_s = mountedLandingPage.find('i')
        expect(i_s.length).toBe(2)
    });

    it('has an image', () => {
        const images = mountedLandingPage.find('Image')
        expect(images.length).toBe(1)
    });

    it('renders the form', () => {
        const forms = mountedLandingPage.find('form')
        expect(forms.length).toBe(1)
    });

    it('renders form input', () => {
        const inputs = mountedLandingPage.find('input')
        expect(inputs.length).toBe(4)
    });

    it('has divs', () => {
        const divs = mountedLandingPage.find('div')
        expect(divs.length).toBe(15)
    });

    it('has links', () => {
        const links = mountedLandingPage.find('Link')
        expect(links.length).toBe(1)
    });

    it('has buttons', () => {
        const buttons = mountedLandingPage.find('Button')
        expect(buttons.length).toBe(2)
    });

    it('has an h1', () => {
        const his = mountedLandingPage.find('h1')
        expect(his.length).toBe(1)
    });
});