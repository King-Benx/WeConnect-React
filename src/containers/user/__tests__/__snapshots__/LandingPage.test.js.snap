// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`These are tests for the registration form for a user renders without crashing 1`] = `
<div
  className="row"
>
  <div
    className="col-lg-4 col-md-4 col-sm-6 col-xs-12 pull-right"
  >
    <div
      className="panel panel-info"
    >
      <div
        className="panel-heading"
      >
        <h3
          className="panel-title"
        >
          Registration Form
        </h3>
      </div>
      <div
        className="panel-body"
      >
        <form
          className="form"
          method="POST"
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="username"
              name="username"
              onChange={[Function]}
              placeholder="Username"
              required="required"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              aria-describedby="email-addon"
              className="form-control"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="johndoe@mail.com"
              required="required"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              aria-describedby="password-addon"
              className="form-control"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              required="required"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              aria-describedby="confirm_password-addon"
              className="form-control"
              id="confirm_password"
              name="confirm_password"
              onChange={[Function]}
              placeholder="Confirm Password"
              required="required"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <button
              className="pull-left btn btn-default"
              disabled={false}
              onClick={[Function]}
              type="reset"
            >
              <i
                className="glyphicon glyphicon-refresh"
              />
               Reset
            </button>
            <button
              className="pull-right btn btn-primary"
              disabled={false}
              type="submit"
            >
              <i
                className="glyphicon glyphicon-plus-sign"
              />
               Register
            </button>
            <div
              className="clearfix"
            />
          </div>
        </form>
        <div
          className="text-center"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Already a member? Login Here!
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="col-lg-8 col-md-8 col-sm-6 col-xs-12 pull-left"
  >
    <div
      className="jumbotron"
    >
      <div
        className="container"
      >
        <h1>
          Welcome to WeConnect
        </h1>
        <p>
          WeConnect provides a platform that brings businesses and individuals together. This platform creates awareness for businesses and gives the users the ability to write reviews about the businesses they have interacted with.
        </p>
        <img
          className="img-responsive"
          src="business-3152586_640.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;
